lib/kafo/kafo_configure.rb

```diff
@@ -366,13 +367,15 @@ def set_options
     # so we accept either allowed args or those that does not start with '-' and are right after
     # accepted argument
     def clamp_app_arguments
-      @allowed_clamp_app_arguments = self.class.app_options.map do |option|
+      @allowed_clamp_app_arguments = self.class.declared_options.map do |option|
         option.switches.map { |s| is_yes_no_flag?(s) ? build_yes_no_variants(s) : s }
```

Comment:
Not for this patch but reading this and the design I started to wonder if we should call the private method [recognised_switches](https://github.com/mdub/clamp/blob/b95498a2c44c48d1703548dc20625ec528b97166/lib/clamp/option/definition.rb#L74-L82) instead. Now that we have defined our own class, we can expose it and drop our copies.

Reply:
Going to track that suggestion here so I do not forget about it -- https://projects.theforeman.org/issues/31175